procurement_auto_create_group_by_product/models/stock_rule.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    auto_create_group_by_product = fields.Boolean(string="Procurement Group by Product")
+
+    def _get_auto_procurement_group(self, product):
+        if self.auto_create_group_by_product:
+            if product.auto_create_procurement_group_ids:
+                return fields.first(product.auto_create_procurement_group_ids)
```

Comment:
Maybe we should add a lock, to ensure there is only one group created in case of auto_by_product?
```suggestion
            if product.auto_create_procurement_group_ids:
                return fields.first(product.auto_create_procurement_group_ids)
            else:
                 # lock to ensure there is only one group created
```

Reply:
@jbaudoux could you merge https://github.com/jbaudoux/stock-logistics-warehouse/pull/1 ?